### DISCUSSION TITLE
Position hold with optical flow/lidar only on iNAV 6.0

### DISCUSSION THREAD

**Discussion by oloendithas** [iNavFlight/inav#8530] [Q&A] [answered]
Hi,

My setup is Foxeer Reaper V2, MS5611 baro, Matek 3901-L0X optical flow/lidar sensor, iNAV 6.0 to support all this.
Opflow and lidar are setup and tested. Lidar actually pretty precise on up to 2 meters. 

So now I'd like to have an indoor navigation. 
I've set:
`set inav_allow_dead_reckoning = ON
set nav_extra_arming_safety = ALLOW_BYPASS (no OFF in latest versions)`
I have combined NAV ALTHOLD/POSHOLD with a SURFACE modifier.
But still unable to arm in NAV POSHOLD, can't bypass the "Nav unsafe" error even with maxed YAW stick. 
I can arm in NAV ALTHOLD, but it behaves weird: it doesn't level after I release the pitch/roll, instead it keeps moving in the lash direction.

Please help?(

**Comment by breadoven** [comment, marked as answer]
You can't arm with POSHOLD selected before arming. `nav_extra_arming_safety `BYPASS doesn't work in this case, it only works if you have Nav modes set up and no GPS fix. It just allows you to Arm without waiting for a fix but still won't let you arm if a Nav mode is actually selected before you try and Arm.

**Reply by oloendithas** [reply]
Oh, thanks so much!

**Reply by Doktordrone** [reply]
Same here, im only using a baro and a Matek 3901-L0X optical flow/lidar sensor. 
No GPS module with iNAV 6.0.
I can`t Arm, navigation is not safe and the INAV is searching for Satellites even with: set nav_extra_arming_safety = ALLOW_BYPASS
The PosHold Mode is programmed to channel 8 and definitely not activated when I try to arm.
If  I delete this PosHold flight mode, the quad works...

Please help !

**Reply by oloendithas** [reply]
As smart people told me, you have to arm is LEVEL and then on the ground switch to PosHold.
